libpromises/evalfunction.c

```diff
@@ -4867,16 +4868,39 @@ static FnCallResult FnCallIPRange(EvalContext *ctx, ARG_UNUSED const Policy *pol
         return FnFailure();
     }
 
-    for (const Item *ip = EvalContextGetIpAddresses(ctx); ip != NULL; ip = ip->next)
+    // we match on VIPADDRESS iff no interfaces or IPs were requested
+    if (NULL == finalargs->next)
```

Comment:
Ah, this was written just before you asked me not to use this style. I'll adjust when I change the function name.